app/graphql/mutations/update_drink.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Mutations
+  class UpdateDrink < BaseMutation
+    description "Updates a drink by id"
+
+    argument :id, ID, required: true
+    argument :ingredients, [Types::IngredientInputType], required: true
+    argument :name, String, required: true
+    argument :steps, String, required: true
+    argument :img_url, String, required: true
+  
+    field :drink, Types::DrinkType
+    field :errors, [String]
+
+    def resolve(id:, ingredients:, name:, steps:, img_url:)
+      drink = Drink.find(id)
+      drink.ingredients.destroy_all
```

Comment:
Have you thought about `accepts_nested_attributes`?

You can allow for destroy in there as well

```
drink_params

{
  ....,
  ingredient_attributes: [
    {
       id: ID,
       name: updating_ingredient,
       ....
    },
    {
       id: '',
       name: creating_ingredient,
       ....
    },
    {
       id: ID,
       name: deleting_ingredient,
      _destroy: '1'
       ....
    },
  ]
}
```

Reply:
https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html